pkg/apis/v1/nodeclaim.go

```diff
@@ -40,21 +40,19 @@ type NodeClaimSpec struct {
 	// +kubebuilder:validation:MaxItems:=100
 	// +required
 	Requirements []NodeSelectorRequirementWithMinValues `json:"requirements" hash:"ignore"`
-	// Resources models the resource requirements for the NodeClaim to launch
-	// +optional
-	Resources ResourceRequirements `json:"resources,omitempty" hash:"ignore"`
```

Comment:
We don't want to drop resources from the NodeClaim. This is still relevant for us understanding when certain resources register. What we wanted to drop was this stanza from the public NodePool OpenAPI coinfig

Reply:
The idea here is to make the NodeClaim only be the CRD that contains the resource field. This is done by creating type `NodeClaimTemplateSpec` as the struct that will hold all shared components between the `NodeClamTemplate` and `NodeClaimSpec`. In my mind, I think this would be the way to go, since the other option is having a script to remove the field. I think scripting should be the last resort, as that introduces additional complexity that will need to reason about